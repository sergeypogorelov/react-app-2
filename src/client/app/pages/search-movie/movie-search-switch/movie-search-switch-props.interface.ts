import { SearchBy } from '../../../core/enums/search-by.enum';
import { CommonProps } from '../../../core/interfaces/common/common-props.interface';

export interface MovieSearchSwitchProps extends CommonProps<{}> {
  value: SearchBy;
}
