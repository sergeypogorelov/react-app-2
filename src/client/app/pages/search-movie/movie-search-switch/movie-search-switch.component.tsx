import React from 'react';
import { connect } from 'react-redux';

import { SearchBy } from '../../../core/enums/search-by.enum';

import { AppState } from '../../../redux/reducers';
import { SwitchProps } from '../../../shared/switch-container/switch/switch-props.interface';
import { MovieSearchSwitchProps } from './movie-search-switch-props.interface';

import { changeSearchBy } from '../../../redux/actions/search-movie-page/search-movie-page';

import { SwitchContainer } from '../../../shared/switch-container/switch-container.component';

class MovieSearchSwitchClass extends React.Component<MovieSearchSwitchProps> {
  render() {
    const switchProps: SwitchProps = {
      currentValue: this.props.value,
      left: {
        label: 'Title',
        value: SearchBy.Title,
      },
      right: {
        label: 'Genre',
        value: SearchBy.Genres,
      },
      onChange: this.handleSwitchChagne,
    };

    return (
      <SwitchContainer switchProps={switchProps}>SEARCH BY</SwitchContainer>
    );
  }

  handleSwitchChagne = (newValue: string) => {
    this.props.dispatch(changeSearchBy(newValue as SearchBy));
  };
}

const mapStateToProps = (state: AppState): MovieSearchSwitchProps => {
  return {
    value: state.searchMoviePage.searchBy,
  };
};

export const MovieSearchSwitch = connect(mapStateToProps)(
  MovieSearchSwitchClass
);
