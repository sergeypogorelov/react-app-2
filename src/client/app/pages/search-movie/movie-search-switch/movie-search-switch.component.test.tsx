import React from 'react';
import renderer from 'react-test-renderer';

import { MovieSearchSwitch } from './movie-search-switch.component';

test('MovieSearchSwitch component should render correctly', () => {
  const tree = renderer.create(<MovieSearchSwitch />).toJSON();
  expect(tree).toMatchSnapshot();
});
