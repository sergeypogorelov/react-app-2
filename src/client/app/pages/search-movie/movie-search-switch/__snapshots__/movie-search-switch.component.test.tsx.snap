// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSearchSwitch component should render correctly 1`] = `
<div
  className="switch-container d-flex align-items-center"
>
  <label
    className="switch-container-label my-0"
  >
    SEARCH BY
  </label>
  <div
    className="switch d-flex"
  >
    <button
      className="switch-item switch-item-left active"
    >
      Title
    </button>
    <button
      className="switch-item switch-item-right"
    >
      Genre
    </button>
  </div>
</div>
`;
